plugins/destination/postgresql/client/migrate.go

```diff
@@ -3,43 +3,125 @@ package client
 import (
 	"context"
 	"fmt"
-	"sort"
 	"strings"
 
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugin-sdk/specs"
 	"github.com/jackc/pgx/v5"
-	"golang.org/x/exp/maps"
 )
 
 const (
-	isTableExistSQL = "select count(*) from information_schema.tables where table_name = $1 and table_schema in (SELECT TRIM(BOTH from unnest(string_to_array(reset_val,','))) FROM pg_settings WHERE name='search_path')"
-
-	// https://wiki.postgresql.org/wiki/Retrieve_primary_key_columns
-	sqlSelectPrimaryKeys = `
-SELECT a.attname as pkey FROM pg_index i       
-JOIN   pg_attribute a ON a.attrelid = i.indrelid
-  AND a.attnum = ANY(i.indkey)
-WHERE  i.indrelid = $1::regclass
-AND    i.indisprimary;
+	// this returns the following table in sorted manner:
+	// +----------------+-------------+-------------+------------+---------------+-----------+---------------------+
+	// | ordinal_position | table_name | column_name | data_type | is_primary_key| not_null  | pk_constraint_name  |
+	// +----------------+-------------+-------------+------------+---------------+-----------+---------------------+
+	// |              1 | users       | id          | bigint     | YES           | true 		 | cq_users_pk 	  	   |
+	// |              2 | users       | name        | text       | NO            | false 	   | 					           |
+	// |              3 | users       | email       | text       | NO            | true 		 | cq_users_pk         |
+	// |              1 | posts       | id          | bigint     | YES           | true 		 | cq_posts_pk			   |
+	// |              2 | posts       | title       | text       | NO            | false 	   | 					           |
+	selectAllTables = `
+ SELECT
+ columns.ordinal_position AS ordinal_position,
+ pg_class.relname AS table_name,
+ pg_attribute.attname AS column_name,
+ pg_catalog.format_type(pg_attribute.atttypid, pg_attribute.atttypmod) AS data_type,
+ CASE 
+		 WHEN contype = 'p' THEN true
+		 ELSE false
+ END AS is_primary_key,
+ CASE 
+		 WHEN pg_attribute.attnotnull THEN true
+		 ELSE false
+ END AS not_null,
+ COALESCE(pg_constraint.conname, '') AS primary_key_constraint_name
+ FROM
+ pg_catalog.pg_attribute
+ INNER JOIN
+ pg_catalog.pg_class ON pg_class.oid = pg_attribute.attrelid
+ INNER JOIN
+ pg_catalog.pg_namespace ON pg_namespace.oid = pg_class.relnamespace
+ LEFT JOIN
+ pg_catalog.pg_constraint ON pg_constraint.conrelid = pg_attribute.attrelid
+ AND pg_constraint.conkey[1] = pg_attribute.attnum
+ AND contype = 'p'
+ INNER JOIN
+ information_schema.columns ON columns.table_name = pg_class.relname AND columns.column_name = pg_attribute.attname AND columns.table_schema = pg_catalog.pg_namespace.nspname
+ WHERE
+ pg_attribute.attnum > 0
+ AND NOT pg_attribute.attisdropped
+ AND pg_catalog.pg_namespace.nspname = '%s'
+ ORDER BY
+ table_name ASC , ordinal_position ASC;
```

Comment:
I think this query is not returning the expected results for tables with composite primary keys.

Here is a table I have locally (`\d aws_s3_bucket_grants`):

```
 aws_s3_bucket_grants
+-----------------+-----------------------------+-----------+
| Column          | Type                        | Modifiers |
|-----------------+-----------------------------+-----------|
| _cq_source_name | text                        |           |
| _cq_sync_time   | timestamp without time zone |           |
| _cq_id          | uuid                        |  not null |
| _cq_parent_id   | uuid                        |           |
| account_id      | text                        |           |
| bucket_arn      | text                        |  not null |
| grantee_type    | text                        |  not null |
| grantee_id      | text                        |  not null |
| grantee         | jsonb                       |           |
| permission      | text                        |  not null |
+-----------------+-----------------------------+-----------+
Indexes:
    "aws_s3_bucket_grants_cqpk" PRIMARY KEY, btree (bucket_arn, grantee_type, grantee_id, permission)
    "aws_s3_bucket_grants__cq_id_key" UNIQUE CONSTRAINT, btree (_cq_id)

Time: 0.040s
```

And here are the results for this table from this query:

```
+------------------+----------------------+-----------------+-----------------------------+----------------+----------+-----------------------------+
| ordinal_position | table_name           | column_name     | data_type                   | is_primary_key | not_null | primary_key_constraint_name |
|------------------+----------------------+-----------------+-----------------------------+----------------+----------+-----------------------------|
| 1                | aws_s3_bucket_grants | _cq_source_name | text                        | False          | False    |                             |
| 2                | aws_s3_bucket_grants | _cq_sync_time   | timestamp without time zone | False          | False    |                             |
| 3                | aws_s3_bucket_grants | _cq_id          | uuid                        | False          | True     |                             |
| 4                | aws_s3_bucket_grants | _cq_parent_id   | uuid                        | False          | False    |                             |
| 5                | aws_s3_bucket_grants | account_id      | text                        | False          | False    |                             |
| 6                | aws_s3_bucket_grants | bucket_arn      | text                        | True           | True     | aws_s3_bucket_grants_cqpk   |
| 7                | aws_s3_bucket_grants | grantee_type    | text                        | False          | True     |                             |
| 8                | aws_s3_bucket_grants | grantee_id      | text                        | False          | True     |                             |
| 9                | aws_s3_bucket_grants | grantee         | jsonb                       | False          | False    |                             |
| 10               | aws_s3_bucket_grants | permission      | text                        | False          | True     |                             |
+------------------+----------------------+-----------------+-----------------------------+----------------+----------+-----------------------------+
```

Notice that the `is_primary_key` column is true only for `bucket_arn`, even though it should be true for `grantee_type`, `grantee_id` and `permission` as well

Reply:
good catch. fixed.